src/execution_plan/ops/op_traverse.c

```diff
@@ -5,17 +5,22 @@
 * modified with the Commons Clause restriction.
 */
 
-#include "op_traverse.h"
 #include <assert.h>
+#include "op_traverse.h"
+#include "../../util/arr.h"
 
 // Updates query graph edge.
 OpResult _Traverse_SetEdge(Traverse *op, Record *r) {
     // Consumed edges connecting current source and destination nodes.
-    Edge *e;
-    if(!Vector_Pop(op->edges, &e)) return OP_DEPLETED;
-
-    Record_AddEntry(r, op->algebraic_expression->edge->alias, SI_PtrVal(e));
+    uint32_t edgeCount = array_len(op->edges);
+    if(!edgeCount) return OP_DEPLETED;
 
+    Edge *e = op->edges + (edgeCount-1);
```

Comment:
Why not do
`Edge *e = array_pop(op->edges)`
and delete line 23?

Reply:
This is because op->edges is an array of structs, array_pop would return an actual struct when I need a pointer.